utils/oomparser/oomparser.go

```diff
@@ -97,28 +96,56 @@ func checkIfStartOfOomMessages(line string) bool {
 	return false
 }
 
-// opens a reader to grab new messages from the Reader object called outPipe
-// opened in PopulateOomInformation.  It reads line by line splitting on
-// the "\n" character.  Checks if line might be start or end of an oom message
-// log. Then the
-// lines are checked against a regexp to check for the pid, process name, etc.
-// At the end of an oom message group, AnalyzeLines adds the new oomInstance to
-// oomLog
-func (self *OomParser) analyzeLines(ioreader *bufio.Reader, outStream chan *OomInstance) {
+// reads the file and sends only complete lines over a channel to analyzeLines.
+// Should prevent EOF errors that occur when lines are read before being fully
+// written to the log. It reads line by line splitting on
+// the "\n" character.
+func analyzeLinesHelper(lineChannel chan string, ioreader *bufio.Reader) {
+	linefragment := ""
 	var line string
 	var err error
 	for true {
-		for line, err = ioreader.ReadString('\n'); err != nil && err == io.EOF; {
+		line, err = ioreader.ReadString('\n')
+		if err == io.EOF {
+			if line != "" {
```

Comment:
how about just `linefragment += line` since if it is empty nothing gets added :)